### PR TITLE
Create models

### PR DESCRIPTION
**Pull request by wuyepabdul**
In this milestone, the following was achieved:

- Created model classes for all Users, Posts, Comments, and Likes entities.
- Set up associations between models.
- Created a method that returns the 3 most recent posts for a given user.
- Created a method that updates the posts counter for a user.
- Created a method that returns the 5 most recent comments for a given post.
- Created a method that updates the comments counter for a post.
-  Created a method that updates the likes counter for a post.
- Created a method called current_user in ApplicationController that returns the first user from a database.

### CODE REVIEW

File: app/models/application_record.rb [hunk 1]
```diff
@@ -1,3 +1,7 @@
 class ApplicationRecord < ActiveRecord::Base
   primary_abstract_class
+
+  def current_user
+    user.first
+  end
```
Comment (bimbolabuari) [inline app/models/application_record.rb]: In order to make current user method data available to all controllers, Kindly move the current_user method to your Application Controller. ✊ 
`app/controllers/application_controller.rb`

File: app/models/comment.rb [hunk 2]
```diff
@@ -0,0 +1,4 @@
+class Comment < ApplicationRecord
+  belongs_to :user
+  belongs_to :post
+end
```
Comment (bimbolabuari) [inline app/models/comment.rb]: As part of the project requirement, Let's add a method that updates the comments counter for a post.

File: app/models/like.rb [hunk 3]
```diff
@@ -0,0 +1,3 @@
+class Like < ApplicationRecord
+  belongs_to :user, :post
+end
```
Comment (bimbolabuari) [inline app/models/like.rb]: As part of the project requirement, Let's add a method that updates the likes counter for a post in the `like` model 🙂 

> image

File: app/models/user.rb [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+  has_many :likes, through: :posts
+  has_many :comments, through: :posts
+
+  def most_recent_posts
+    posts.last(3)
+  end
+
+  def increment_posts_counter
+    user.posts_counter += 1
+  end
```
Comment (bimbolabuari) [inline app/models/user.rb]: I suggest we move this to the `post` model as a method that updates the posts counter for a user. What do you think? 🤔 

> image

File: app/models/post.rb [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+    comments.last(5)
+  end
+
+  def increment_likes_counter
+    posts.likes_counter += 1
+  end
+
+  def increment_comments_counter
+    posts.comments_counter += 1
+  end
```
Comment (bimbolabuari) [inline app/models/post.rb]: For the `post` model, Its is required to create just two methods 👍 
> image

**Review by bimbolabuari — CHANGES_REQUESTED** [review]
Hi @wuyepabdul , 

You have done a good job so far 🚀  

## Highlights 🎉 

- Create model classes for all entities as shown in the [ERD diagram.](https://github.com/microverseinc/curriculum-rails/blob/main/blog-app/images/blog_app_erd.png)
- Set up associations between models.

There are some issues that you still need to work on to go to the next project but you are almost there! 🎉 
<FEEL FREE TO CHANGE THIS MESSAGE TO YOUR PERSONAL WORDS>


## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

--------


_Reviewed by  @bimbolabuari_

**Review by Emmyn5600 — APPROVED** [review]
Hi @wuyepabdul

Wooow 👏, Your project is complete!  There is nothing else to say other than... it's time to merge it. 👏

Congratulations! 🎉 

##  Optional suggestion

- n/a

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Review by Emmyn5600 — APPROVED** [review]
Hi @wuyepabdul

Wooow 👏, Your project is complete!  There is nothing else to say other than... it's time to merge it. 👏

Congratulations! 🎉 

##  Optional suggestion

- n/a

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Comment by wuyepabdul** [comment]
Thanks for the review @Emmyn5600 and @bimbolabuari

**Pull request merged**